### ISSUE TITLE
Can't build the installer for Ubuntu because of spaces in the path

### ISSUE THREAD

**Issue by j0j0h** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3285]
I can start the app using `npm start` but with `npm run make:deb` I get some of the following:

```
Packaging app for platform linux ia32 using electron v2.0.2  
You are using the deprecated electron-rebuild API, please switch to using the options object instead  
Attempted to build a module with a space in the path  
See https://github.com/nodejs/node-gyp/issues/65#issuecomment-368820565 for reasons why this will not work  
[21:17:24] 'package:linux:32' errored after 18 s  
[21:17:24] Error: node-gyp does not support building modules with spaces in their path, tried to build: /tmp/electron-packager/linux-ia32/Google Play Music Desktop Player-linux-ia32/resources/app/node_modules/dbus  
    at Rebuilder.<anonymous> (/home/jojoh/Documents/muz/node_modules/electron-rebuild/src/rebuild.ts:233:13)  
    at Generator.next (<anonymous>)  
    at fulfilled (/home/jojoh/Documents/muz/node_modules/electron-rebuild/lib/src/rebuild.js:4:58)  
[21:17:24] 'deb:linux' errored after 23 s  
[21:17:24] Error in plugin 'run-sequence(package:linux:32)'  
Message:  
    node-gyp does not support building modules with spaces in their path, tried to build: /tmp/electron-packager/linux-ia32/Google Play Music Desktop Player-linux-ia32/resources/app/node_modules/dbus  
Stack:  
Error: node-gyp does not support building modules with spaces in their path, tried to build: /tmp/electron-packager/linux-ia32/Google Play Music Desktop Player-linux-ia32/resources/app/node_modules/dbus  
    at Rebuilder.<anonymous> (/home/jojoh/Documents/muz/node_modules/electron-rebuild/src/rebuild.ts:233:13)  
    at Generator.next (<anonymous>)  
    at fulfilled (/home/jojoh/Documents/muz/node_modules/electron-rebuild/lib/src/rebuild.js:4:58)  
Attempted to build a module with a space in the path  
See https://github.com/nodejs/node-gyp/issues/65#issuecomment-368820565 for reasons why this will not work  
Attempted to build a module with a space in the path  
See https://github.com/nodejs/node-gyp/issues/65#issuecomment-368820565 for reasons why this will not work  
Attempted to build a module with a space in the path  
See https://github.com/nodejs/node-gyp/issues/65#issuecomment-368820565 for reasons why this will not work  
npm ERR! code ELIFECYCLE  
npm ERR! errno 1  
npm ERR! google-play-music-desktop-player@4.6.1 make:deb: `gulp make:linux:deb`  
npm ERR! Exit status 1  
npm ERR!   
npm ERR! Failed at the google-play-music-desktop-player@4.6.1 make:deb script.  
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.  
  
npm ERR! A complete log of this run can be found in:  
npm ERR!     /home/jojoh/.npm/_logs/2018-07-22T18_17_24_747Z-debug.log  

```

Seems like node-gyp can't read file paths with spaces, others with the same problem on other projects have only had to remove spaces from their usernames. But I wouldn't know how to make these tmp paths not have spaces in them.

edit: Added more stuff from the terminal in case it might help.

[2018-07-22T17_30_37_941Z-debug.log](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/2217252/2018-07-22T17_30_37_941Z-debug.log)

**Comment by jostrander** [comment]
Interesting. CI manages to build fine so I'll see if I can spot the difference.

**Comment by jostrander** [comment]
I can't reproduce this when building. We specifically use yarn for deps and lock file. It's possible you have the wrong dependencies due to not adhering to the lock file?

**Comment by j0j0h** [comment]
Thanks, I'll have a look at that when I get home.

**Comment by j0j0h** [comment]
So I deleted my gpmdp folder and started fresh, following the dev faq.
I started with
`git clone https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-.git gpmdp
cd gpmdp`
Then installed yarn and got the following
`jojoh@jojoh-ThinkPad-X230:~/Documents/gpmdp$ npm i yarn  
npm notice created a lockfile as package-lock.json. You should commit this file.  
yarn@1.7.0  
added 1 package and audited 1 package in 1.741s  
found 0 vulnerabilities`
So this seemed to create the lock file I should follow if I understand correctly. Then I followed up with the rest of the faqs commands, I could start the app with `npm start` but `npm run make:deb` still spit out the same errors as in OP.
I installed all the dependencies through apt-get as in the FAQ, along with build-essential.
I'm using the following versions of node, npm and yarn
`  jojoh@jojoh-ThinkPad-X230:~$ node --version && npm --version && yarn --version  
v10.7.0  
6.1.0  
1.7.0  `So I deleted my gpmdp folder and started fresh, following the dev faq.
I started with
`git clone https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-.git gpmdp
cd gpmdp`
Then installed yarn and got the following
`jojoh@jojoh-ThinkPad-X230:~/Documents/gpmdp$ npm i yarn  
npm notice created a lockfile as package-lock.json. You should commit this file.  
yarn@1.7.0  
added 1 package and audited 1 package in 1.741s  
found 0 vulnerabilities`
So this seemed to create the lock file I should follow if I understand correctly. Then I followed up with the rest of the faqs commands, I could start the app with `npm start` but `npm run make:deb` still spit out the same errors as in OP.
I installed all the dependencies through apt-get as in the FAQ, along with build-essential.
I'm using the following versions of node, npm and yarn
`  jojoh@jojoh-ThinkPad-X230:~$ node --version && npm --version && yarn --version  
v10.7.0  
6.1.0  
1.7.0  `

**Comment by j0j0h** [comment]
I successfully compiled it! I started digging through the scripts and config files trying to perhaps remove the spaces from that temp folder mentioned, came upon docker-compose-linux-build.yml in the gpmdp root folder, in it is a command to run ``npm run make:deb:64`` , which I'm actually pretty sure to have ran sometime in the past, but this time it compiled.

So the reason could also be that I installed nodejs with [NVM](https://github.com/creationix/nvm/blob/master/README.md#installation), and installed a few packages globally such as gulp, yarn, electron and node-gyp. Not sure if those global packages had anything to do with it, but installing through NVM I could bypass some permission issues certain npm modules had. Case closed.
